src/main/scala/com/gu/riffraff/artifact/RiffRaffArtifact.scala

```diff
@@ -36,19 +33,13 @@ object RiffRaffArtifact extends AutoPlugin {
 
     lazy val riffRaffManifestFile = settingKey[String]("Filename of the build manifest for RiffRaff")
     lazy val riffRaffManifestProjectName = settingKey[String]("Project name for the manifest RiffRaff uses to describe a build")
-    lazy val riffRaffBuildIdentifier = settingKey[String]("Identifier for a particular build of a project")
```

Comment:
I'm all up for removing keys, but I'm wondering if these aren't used in specific cases when we really need to override something. Maybe @sihil would know?

Otherwise, I'm happy to see these gone

Reply:
We currently set a bunch of these in a LOT of places. https://github.com/search?l=&q=riffRaffBuildIdentifier&ref=advsearch&type=Code&utf8=%E2%9C%93.

However, it is largely copy and pasted code for Travis or CircleCI etc. I would suggest expending the effort of automatic detection of BuildInfo for these CI platforms and bumping the major version number when we release with a note telling people they should be able to delete their CI specific logic.

Reply:
Thanks very much for that - I had naively assumed that that (automatic detection) was what was happening anyway, based on the last few lines of BuildInfo.

I'll take a look at adding auto-detection for CircleCI, Travis, etc.

Reply:
I've added BuildInfo implementations for automatic detection within Circle CI and Travis platforms.